client/src/sections/Fridge/Sidebar/Sidebar.tsx

```diff
@@ -9,13 +9,18 @@ export default function Sidebar() {
     "Meats", "Fruits", "Veggies"
   ])
 
```

Comment:
Again, lets move all the logic above the rendering part. Some of these changes are an enhancement of the changes I requested on the previous PR.

```
const [categories, setCategories] = useState([
   "Meats", "Fruits", "Veggies"
])
const [input, setInput] = useState("")

const handleDeleteCategory = (categoryToDelete: string) => {
  setCategories(categories.filter((category) => {return category !== categoryToDelete}));
}

const handleAddCategory = (categoryToAdd: string) => {
   setCategories([...categories, categoryToAdd]);
}

const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
   setInput(event.target.value)
}

const categoriesMarkup = categories.map((category, i) => {
   return (
     <div className="category" key={i} tabIndex={1}>
        {category}
        <span className="close" onClick={() => handleDeleteCategory(category)}>
        x
        </span>
     </div>
   );
})

return (
     <div className='sidebar'>
       {categoriesMarkup}
       <div className="category" onClick={() => handleAddCategory(input)}>+</div>
       <div className="popup">
          <input type="text"  onChange={handleInput}/>
       </div>
    </div>
);
```